src/bun.js/webcore/request.zig

```diff
@@ -818,11 +840,60 @@ pub const Request = struct {
         allocator: std.mem.Allocator,
         globalThis: *JSGlobalObject,
         preserve_url: bool,
+        req_jsvalue: JSC.JSValue,
     ) void {
         _ = allocator;
         this.ensureURL() catch {};
 
-        const body = InitRequestBodyValue(this.body.value.clone(globalThis)) catch {
+        const body = InitRequestBodyValue(
+            switch (this.body.value.clone(globalThis)) {
+                .Empty => brk: {
+                    const stream = stream: {
+                        if (Request.bodyGetCached(req_jsvalue)) |existing_stream_value| {
+                            const existing_stream = JSC.WebCore.ReadableStream.fromJS(existing_stream_value, globalThis) orelse break :stream null;
+                            if (existing_stream.isDisturbed(globalThis)) {
+                                break :stream null;
+                            }
+
+                            break :stream existing_stream;
+                        }
+
+                        if (this.body.value.Locked.readable.get()) |stream| {
+                            if (stream.isDisturbed(globalThis)) {
+                                break :stream null;
+                            }
+
+                            break :stream stream;
+                        }
+
+                        if (this.body.value.Locked.promise == null) {
+                            const stream_value = this.body.value.toReadableStream(globalThis);
+                            break :stream JSC.WebCore.ReadableStream.fromJS(stream_value, globalThis);
+                        }
+
+                        break :stream null;
+                    };
+
+                    if (stream) |s| {
+                        if (s.tee(globalThis)) |teed| {
+                            Request.bodySetCached(req_jsvalue, globalThis, teed[0].value);
+
+                            teed[1].value.protect();
```

Comment:
This `protect` likely causes a memory leak when used because the Strong below does the job of the protect